pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -168,6 +169,15 @@ func (nc *NamespaceController) namespaceChange(ns *v1.Namespace) error {
 // When a config map is changed, merge the data into the configmap
 func (nc *NamespaceController) configMapChange(cm *v1.ConfigMap) error {
 	if err := k8s.UpdateDataInConfigMap(nc.client, cm.DeepCopy(), nc.getData()); err != nil {
+		// configMap may be already deleted
```

Comment:
I think we can simply modify the UpdateFunc like DeleteFunc instead of breaking logic in configMapChange.

Maybe like this.
```
UpdateFunc: func(_, obj interface{}) {
	...
        ...
	c.queue.Push(func() error {
		ns, err := c.namespaceLister.Get(cm.Namespace)
		if err != nil {
			// namespace is deleted before
			if apierrors.IsNotFound(err) {
				return nil
			}
			return err
		}
		// If the namespace is terminating, we may get into a loop of trying to update the configmap back
		// We should make sure the namespace still exists
		if ns.Status.Phase != v1.NamespaceTerminating {
			return c.configMapChange(cm)
		}
		return nil
	})
}
```

Reply:
Yes, it may be better to combine all the xxxFuncs to same codes.

Reply:
I use a new way to handle these events